src/st/dre/14774.c

```diff
@@ -1,169 +1,74 @@
-            self->unk19 = 0;
-        }
-        AnimateEntity(D_80180794, self);
-
-        entity = self->ext.generic.unk9C;
-        if (entity->ext.generic.unk84.U8.unk1 != 0) {
-            self->unk19 = 0;
-            self->step++;
-        }
-        if (entity->flags & 0x100) {
-            DestroyEntity(self);
-        }
-        break;
-
-    case 2:
-        temp_s0 = (self->params << 0xA) + 0x200;
-        self->velocityX = rcos(temp_s0) * 0x38;
-        self->velocityY = rsin(temp_s0) * 0x38;
-        self->ext.generic.unkA2 = temp_s0;
-        self->ext.generic.unk80.modeS16.unk0 = 128;
-        self->step++;
-
-    case 3:
-        AnimateEntity(D_80180794, self);
-        MoveEntity();
-        temp_v0 = func_8019AF08(self, g_Entities);
-        temp_s0 = func_8019AF88(0x10, self->ext.generic.unkA2, temp_v0);
-        self->velocityX = rcos(temp_s0) * 0x38;
-        self->velocityY = rsin(temp_s0) * 0x38;
-        self->ext.generic.unkA2 = temp_s0;
-
-        if (self->hitFlags & 0x80) {
-            self->step = 4;
-        }
+void func_80194774(void) {
+    D_801A3EE4 = 2;
+    D_801A3EE2 = 2;
+    D_801A3EE6 = 0;
+    D_801A3EEA = 0;
+    D_801A3EEE = 0;
+    D_801A3EEF = 8;
+    D_801A3EDE = D_801A3EE0 + 20;
+}
 
-        if (--self->ext.generic.unk80.modeS16.unk0 == 0) {
-            self->step = 4;
-        }
-        break;
+s32 func_801947C8(s32 arg0) {
+    Primitive* prim;
+    s16 firstPrimIndex;
 
-    case 4:
-        self->flags |= FLAG_DESTROY_IF_OUT_OF_CAMERA;
-        AnimateEntity(D_80180794, self);
-        MoveEntity();
-        break;
+    firstPrimIndex = g_api.AllocPrimitives(PRIM_SPRT, 7);
+    D_801A3F10[0] = firstPrimIndex;
+    if (firstPrimIndex == -1) {
+        D_801A3F10[0] = 0;
+        return 0;
     }
-}
+    D_801A3ED8 = arg0;
+    D_801A3F14 = 0;
+    D_801A3F0C = -1;
+    D_801A3F08 = -1;
+    func_80194774();
 
-// Extending wing spike from succubus ID 0x1E
-void EntitySuccubusWingSpike(Entity* self) {
-    s32 temp_s2;
-    s16 var_s0;
-
-    if (D_80180664 & 2) {
-        DestroyEntity(self);
-        return;
+    if (prim && prim) { // !FAKE
     }
 
-    switch (self->step) {
-    case 0:
-        InitializeEntity(D_801804E8);
-        self->unk19 = 4;
-        self->animCurFrame = 0;
-        var_s0 = D_801807F0[self->params];
-        self->rotAngle = var_s0;
-        self->unk19 |= 1;
-        self->unk1A = 0x100;
-        CreateEntityFromEntity(0x1F, self, &self[1]);
-        self[1].facing = self->facing;
-        self[1].params = self->params;
-        self[1].rotAngle = self->rotAngle;
-
-    case 1:
-        if (self->ext.generic.unk9C->ext.generic.unk84.U8.unk1 != 0) {
-            self->step++;
-        }
-        break;
+    prim = D_801A3EF0[0] = &g_PrimBuf[D_801A3F10[0]];
 
-    case 2:
-        self->animCurFrame = 85;
-        self->unk1A += 0x40;
-        if (self->unk1A > 0x600) {
-            self->unk1A = 0x600;
-        }
+    prim->blendMode = 8;
```

Comment:
```suggestion
    prim->blendMode = BLEND_VISIBLE;
```
for all the `blendMode = 8`